### PR TITLE
raftstore-v2: schedule raft tick after apply snapshot 

### LINKED ISSUE #14548: [Dynamic Regions] raft tick may be missing
## Bug Report

In raftstore-v2, when a raft node starts, if there's a noticable gap between commit index and apply index, we deplay scheduling raft tick in order to avoid it being a leader and thus block the raft activities during apply catching up.
In this case, the tick is scheduled when the apply index catches up the commit index in `on_apply_res`.  However, the peer may apply a snapshot so that it will not apply any raft logs until new raft logs are appended. But before that time, there's no raft tick in this peer which could lead some problems
For instance, say we have 3 nodes and node 1 falls in this situation, and network partition occurs leading to [1, 2], [3].
The region will be unavailable during the whole network partition. Just like https://github.com/tikv/tikv/issues/14532.

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #14548, #14532

What's Changed:

```commit-message
schedule raft tick after apply snapshot 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]
rest LGTM

File: tests/failpoints/cases/test_snap.rs [hunk 1]
```diff
@@ -900,3 +900,36 @@ fn test_snapshot_recover_from_raft_write_failure_with_uncommitted_log() {
         cluster.must_put(format!("k1{}", i).as_bytes(), b"v1");
     }
 }
+
+#[test]
+fn test_issue_14548() {
```
[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply tests/failpoints/cases/test_snap.rs:905]: Any suggestions for the name of the test?

[the remark this answers is not in the document]
Comment (tonyxuqqi) [reply tests/failpoints/cases/test_snap.rs:905]: I think it's better to rename the test with the test function. 
In this case, maybe test_write_after_network_partition

File: components/raftstore-v2/src/operation/ready/apply_trace.rs [hunk 2]
```diff
@@ -404,6 +405,13 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
             }
         };
         apply_state.set_applied_index(applied_index);
+        let mut reset_apply_index = || {
+            // Make node reply from start.
+            fail_point!("RESET_APPLY_INDEX_WHEN_RESTART", |_| {
+                apply_state.set_applied_index(5);
+            });
+        };
+        reset_apply_index();
```
Comment (tonyxuqqi) [inline components/raftstore-v2/src/operation/ready/apply_trace.rs:414]: why not calling ```
"  fail_point!("RESET_APPLY_INDEX_WHEN_RESTART", |_| {
                apply_state.set_applied_index(5);
            });" 
```directly?

**Review by tonyxuqqi — APPROVED** [review]
Some minor issues are better to address before the merge.

File: components/raftstore-v2/src/operation/ready/apply_trace.rs (hunk 2 above)
Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/ready/apply_trace.rs:414]: The failpoint will return.

File: tests/failpoints/cases/test_snap.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (SpadeA-Tang) [reply tests/failpoints/cases/test_snap.rs]: Done

**Pull request merged**